### ISSUE TITLE
Cannot deploy a React outlook add-in created with Yo Office to our own ubuntu server?

### ISSUE THREAD

**Issue by eurog33k** [OfficeDev/generator-office#436]
If I deploy a Yo Office generated Outlook add-in that uses the Jquery  framework to our Ubuntu server, it works.

Unfortunately since I want to use Fabric React since it has more components than FabricJS I developed my add-in with the React framework. After building and deploying it to our Ubuntu server, the add-in made with React does not work. When I check the element inspector in Outlook, it indicates that It can't find the variable React.
(it's in the code as import * as React from 'react';)

Just to make sure it wasn't an error with my code I deployed an empty Outlook add-in generated with yo Office using the React Framework and it gives the exact same error.
The exact steps I took can be found here https://stackoverflow.com/questions/55420517/how-do-i-deploy-an-outlook-add-in-created-with-yo-office-to-our-own-ubuntu-serve

Am I doing something wrong, or is something broken with the build process?

**Comment by SPDoctor** [comment]
This is because the webpack configuration for the production build marks React and React-DOM as external. You can either load them explicitly with script links in the HTML page, or go to /config/webpack.prod.js and remove the externals section:
```
  externals: {
        'react': 'React',
        'react-dom': 'ReactDOM'
    },

```

Doing the latter will cause webpack to load React into the bundle.

**Comment by eurog33k** [comment]
This is apparently because the webpack configuration for the production build marks React and ReactDOM as external.
This was not obvious to me at all and I did not find this in any documentation either.
As a matter of fact how to deploy to your own server would be a nice addition to the docs, because the fact that I had to "npm run build" wasn't obvious either. Most documentation I found online people installed the whole thing including the dev tools on their servers. That shouldn't be necessary at all. So I guess the issue is more with the documentation than with this project. Could someone switch it to the documentation project please.
Thanks in advance.

**Comment by TCourtneyOwen** [comment]
Hi Dirk, thanks for bringing this our attention and sorry that you are having problems with the React project template.  I will look into this today and see if we can make improvements to the default template or documentation so that things are less confusing.  I will keep you posted.

-Courtney

**Comment by SPDoctor** [comment]
Have to decide whether, in the template, to bundle React or add a script link in the HTML file. Currently the generator produces a solution that can't be deployed without changes.

My preferred option would be to bundle, which is simpler and makes it easier to guarantee you have the correct React version. This will add about 3Mb to the bundle, but the client has to load this one way or another. OTOH the script link has the advantage that you have the option of using a public CDN.

**Comment by TCourtneyOwen** [comment]
@Rick-Kirkham @Reezaali @kbrandl Looping in a few of our Writers on this issue to see about making our deployment documentation clearer

**Comment by Rick-Kirkham** [comment]
I believe that including it in the bundle is the industry standard. But we should probably say that, when the add-in is ready for production deployment, devs have a choice of the two ways.

**Comment by SPDoctor** [comment]
@TCourtneyOwen It's not just the documentation - the Yeoman generator produces a solution that can't be deployed without changes. I suggest remove the externals from /config/webpack.prod.js in the template so React is included in the bundle, and document how to use CDN if that is the developer's preference.

**Comment by TCourtneyOwen** [comment]
@SPDoctor I just looked at your PR to remove the externals from  /config/webpack.prod.js and all looks well. After running 'npm run build', I was able to simply copy the contents of the dist folder to %inetpub\wwwroot (I am using IIS), and update the urls in manifest.xml to point to the IIS instance.

**Comment by SPDoctor** [comment]
@TCourtneyOwen Great! Thanks for responding so quickly :-)

**Comment by TCourtneyOwen** [comment]
Just published out the new version of generator-office that fixes this issue (1.1.27).

Thanks,

Courtney
